CCP_App/src/main/java/com/example/demo2/config/SecurityConfig.java

```diff
@@ -0,0 +1,71 @@
+ *  DDDD    SSSS    A   A       Telefax: +49 (0)2408 / 9492-92          *
+ *                                                                      *
+ *                                                                      *
+ *  (c) Copyright by DSA - all rights reserved                          *
+ *                                                                      *
+ ************************************************************************
+ *
+ * Author : Riyaardh Adam
+ * Created: 2023/02/24 09:09
+ *
+ ***********************************************************************/
+package com.example.demo2.config;
+
+import org.keycloak.adapters.springboot.KeycloakSpringBootConfigResolver;
+import org.keycloak.adapters.springsecurity.KeycloakConfiguration;
+import org.keycloak.adapters.springsecurity.authentication.KeycloakAuthenticationProvider;
+import org.keycloak.adapters.springsecurity.config.KeycloakWebSecurityConfigurerAdapter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.core.authority.mapping.SimpleAuthorityMapper;
+import org.springframework.security.core.session.SessionRegistry;
+import org.springframework.security.core.session.SessionRegistryImpl;
+import org.springframework.security.web.authentication.session.RegisterSessionAuthenticationStrategy;
+import org.springframework.security.web.authentication.session.SessionAuthenticationStrategy;
+
+@KeycloakConfiguration
+@EnableGlobalMethodSecurity(jsr250Enabled = true)
+@Import(KeycloakSpringBootConfigResolver.class)
+public class SecurityConfig extends KeycloakWebSecurityConfigurerAdapter{
+	/**
+	 * Registers the KeycloakAuthenticationProvider with the authentication manager.
+	 */
+	@Autowired
+	public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
+		KeycloakAuthenticationProvider authenticationProvider = new KeycloakAuthenticationProvider();
+		authenticationProvider.setGrantedAuthoritiesMapper(new SimpleAuthorityMapper());
+		auth.authenticationProvider(authenticationProvider);
+	}
+
+	/**
+	 * Defines the session authentication strategy.
+	 */
+	@Bean
+	@Override
+	protected SessionAuthenticationStrategy sessionAuthenticationStrategy() {
+		return new RegisterSessionAuthenticationStrategy(buildSessionRegistry());
+	}
+
+	@Bean
+	protected SessionRegistry buildSessionRegistry() {
+		return new SessionRegistryImpl();
+	}
+
+	@Override
+	protected void configure(HttpSecurity http) throws Exception	{
+		super.configure(http);
+		http
+				.authorizeRequests()
+				.anyRequest().permitAll();
```

Comment:
Please make a test on postman if you make an unauthorized request to your service that you get a 401 response. the permitAll() makes me curious if its working as intended but I dont know the HttpSecurity class indepth